django/db/models/lookups.py

```diff
@@ -274,6 +274,18 @@ def process_rhs(self, compiler, connection):
                 )
         return super().process_rhs(compiler, connection)
 
+    def as_sql(self, compiler, connection):
+        # Avoid comparison against direct rhs if lhs is a boolean value.
+        # That turns "boolfield__exact=True" into "WHERE boolean_field"
+        # instead of "WHERE boolean_field = True" when allowed.
+        if (isinstance(self.rhs, bool) and
+                getattr(self.lhs, 'conditional', False) and
+                connection.ops.conditional_expression_supported_in_where_clause(self.lhs)):
+            lhs_sql, params = self.process_lhs(compiler, connection)
+            template = '%s' if self.rhs else 'NOT %s'
+            return template % lhs_sql, params
+        return super().as_sql(compiler, connection)
```

Comment:
`expressions.tests.BasicExpressionsTests.test_filtering_on_q_that_is_boolean` is failing on Oracle after this change. This is caused by changing a type of `expression.expression` for an `ExpressionWrapper`, it's a `Q` expression before resolving but here it's already a `models.lookups.GreaterThan` lookup (see [conditional_expression_supported_in_where_clause](https://github.com/django/django/blob/e9def97d1095efed15a109d82fe0498ebd56fa04/django/db/backends/oracle/operations.py#L638-L639)).

Reply:
It was only a matter of checking against what `Q` resolves to instead; a `WhereNode`.